### DISCUSSION TITLE
Multiple instances per page

### DISCUSSION THREAD

**Discussion by felixhuttmann** [vikejs/vike#905] [Discussion]
We have a page where we want to use multiple vue app instances with pre-rendering. The page is assembled at runtime from multiple fragments that come from different services using server-side includes.
1) 
```
Uncaught (in promise) Error: [vite-plugin-ssr][Wrong Usage] Both `vite-plugin-ssr@0.4.42` and `vite-plugin-ssr@0.4.123` loaded. Only one version should be loaded.
```
We would like to have the ability to have different versions because the services are independently released.

2) Even when the version is the same, there are errors
```
assertSingleInstance.js?v=7ffc53a0:89 Error: [vite-plugin-ssr][Warning] Two vite-plugin-ssr client runtime instances are being loaded. Make sure your client-side code doesn't include(/bundle) vite-plugin-ssr twice. (In order to reduce the size of your client-side JavaScript bundles.)
```
and due to an exception
```
chunk-Q5TZPECP.js?v=7ffc53a0:224 Uncaught (in promise) TypeError: Object.defineProperty called on non-object
    at Function.defineProperty (<anonymous>)
    at def (chunk-Q5TZPECP.js?v=7ffc53a0:224:10)
    at markRaw (chunk-Q5TZPECP.js?v=7ffc53a0:1053:3)
    at Proxy.data (app.ts:31:13)
    at applyOptions (chunk-Q5TZPECP.js?v=7ffc53a0:3902:30)
    at finishComponentSetup (chunk-Q5TZPECP.js?v=7ffc53a0:6849:5)
    at handleSetupResult (chunk-Q5TZPECP.js?v=7ffc53a0:6807:3)
    at setupStatefulComponent (chunk-Q5TZPECP.js?v=7ffc53a0:6780:7)
    at setupComponent (chunk-Q5TZPECP.js?v=7ffc53a0:6724:36)
    at mountComponent (chunk-Q5TZPECP.js?v=7ffc53a0:5530:7)
```
the components do not proceed to load.

Is it at all on the table to have two vite-plugin-ssr instancees on one page? Without SSR, various websites report that it is possible to have multiple vue instances on the same page just fine.

**Comment by brillout** [comment]
> and due to an exception [...]  the components do not proceed to load.

```
    at Function.defineProperty (<anonymous>)
    at def (chunk-Q5TZPECP.js?v=7ffc53a0:224:10)
```

Is that user land or VPS land?

> Is it at all on the table to have two vite-plugin-ssr instancees on one page? 

It isn't trivial to achieve and it isn't a top priority. Do other frameworks like Nuxt support that?

**Reply by felixhuttmann** [reply]
Thank you for your rapid reply.

> Is that user land or VPS land?

I am not sure what you mean with user vs VPS. Do you mean browser vs server? It is in the browser console.

> Do other frameworks like Nuxt support that?

I do not know if other frameworks like nuxt support it.

I guess we will have to go without prerendering for now.

**Reply by brillout** [reply]
Whether the code at the beginning of the errror stack trace originates from your app source code of from VPS's source code.

I don't know if your company is open for sponsoring, but if it is then I'd be up to increase the priority of this. If there is interest on your side feel free to PM me on Discord so we can discuss a solution.
